### ISSUE TITLE
Jetpack.com: prices in product cards are not announced correctly by screen readers

### ISSUE THREAD

**Issue by jkguidaven** [Automattic/jetpack#29740]
## What's the issue
In the product cards on Jetpack.com, prices can't be read properly by screen readers. They read a series of symbols and digits instead. Besides, there's no differentiation between the discounted and the actual prices.

## How to reproduce the issue
- Visit https://jetpack.com/
- Browse the page using a screen reader
- Notice that:
  -  prices are announced as a series of symbols and digits
  - regular and discounted prices are not differentiated

## Expected behavior
Screen readers should say something like `1 dollar instead of 10 dollars, for the first month, billed yearly`.

## Captures

https://user-images.githubusercontent.com/1620183/227224573-66c52a22-3b14-4edd-9f2a-41d3a51ea014.mov

_Tested with VoiceOver on macOS/Chrome_

**Comment by jkguidaven** [comment]
diff ID: D106396-code

**Comment by jkguidaven** [comment]
diff has been deployed.
